src/unicycle_2d/ignition.cpp

```diff
@@ -0,0 +1,257 @@
+  if (reset_optimizer && !params_.reset_service.empty())
+  {
+    auto srv = std_srvs::Empty();
+    if (!reset_client_.call(srv))
+    {
+      // The reset() service failed. Propagate that failure to the caller of this service.
+      ROS_ERROR_STREAM("Failed to call the " << reset_client_.getService() << " service.");
+      return false;
+    }
+  }
+
+  const auto& stamp = pose.header.stamp;
+
+  // Create variables for the full state.
+  // The initial values of the pose are extracted from the provided PoseWithCovarianceStamped message.
+  // The remaining dimensions are provided as parameters to the parameter server.
+  auto position = fuse_variables::Position2DStamped::make_shared(stamp, device_id_);
+  position->x() = pose.pose.pose.position.x;
+  position->y() = pose.pose.pose.position.y;
+  auto orientation = fuse_variables::Orientation2DStamped::make_shared(stamp, device_id_);
+  orientation->yaw() = fuse_core::getYaw(pose.pose.pose.orientation.w,
+                                         pose.pose.pose.orientation.x,
+                                         pose.pose.pose.orientation.y,
+                                         pose.pose.pose.orientation.z);
+  auto linear_velocity = fuse_variables::VelocityLinear2DStamped::make_shared(stamp, device_id_);
+  linear_velocity->x() = params_.initial_state[0];
+  linear_velocity->y() = params_.initial_state[1];
+  auto angular_velocity = fuse_variables::VelocityAngular2DStamped::make_shared(stamp, device_id_);
+  angular_velocity->yaw() = params_.initial_state[2];
+  auto linear_acceleration = fuse_variables::AccelerationLinear2DStamped::make_shared(stamp, device_id_);
+  linear_acceleration->x() = params_.initial_state[3];
+  linear_acceleration->y() = params_.initial_state[4];
+
+  // Create the covariances for each variable
+  // The pose covariances are extracted from the provided PoseWithCovarianceStamped message.
+  // The remaining covariances are provided as parameters to the parameter server.
+  auto linear_velocity_cov = fuse_core::Matrix2d();
+  linear_velocity_cov << params_.initial_sigma[0] * params_.initial_sigma[0], 0.0,
+                         0.0, params_.initial_sigma[1] * params_.initial_sigma[1];
+  auto angular_velocity_cov = fuse_core::Matrix1d();
+  angular_velocity_cov << params_.initial_sigma[2] * params_.initial_sigma[2];
+  auto linear_acceleration_cov = fuse_core::Matrix2d();
+  linear_acceleration_cov << params_.initial_sigma[3] * params_.initial_sigma[3], 0.0,
+                             0.0, params_.initial_sigma[4] * params_.initial_sigma[4];
+
+  // Create absolute constraints for each variable
+  auto position_constraint = fuse_constraints::AbsolutePosition2DStampedConstraint::make_shared(
+    *position,
+    fuse_core::Vector2d(position->x(), position->y()),
+    position_cov);
+  auto orientation_constraint = fuse_constraints::AbsoluteOrientation2DStampedConstraint::make_shared(
```

Comment:
We are cheating here a little bit by creating separate position and orientation constraints. This prevents us from including the position-orientation correlation terms from the PoseWithCovarianceStamped message. I could switch this to an AbsolutePose2DStampedConstraint if we feel like that is important.

Reply:
I say leave it this way. I don't think I've ever seen anyone set the off-diagonal terms in a `SetPose` service call, or from the parameters.